crates/sc-proof-of-time/src/time_keeper.rs

```diff
@@ -225,11 +234,10 @@ where
         let ret = self.pot_state.on_proof_from_peer(sender, &proof);
         let elapsed = start_ts.elapsed();
 
-        if let Err(err) = ret {
-            trace!("time_keeper::on gossip: {err:?}, {sender}");
+        if let Err(error) = ret {
+            trace!(%error, %sender, "On gossip");
         } else {
-            trace!("time_keeper::on gossip: {proof}, time=[{elapsed:?}], {sender}");
-            self.gossip.gossip_message(proof.encode());
```

Comment:
This is needed .. this is a difference between time keeper and node client

Reply:
Hm... I'm sorry, I think I missed it when copy-pasting from one file to another :confused: 

Why though? The whole point of gossip engine is to handle gossip, I don't think you need to manually re-gossip what you have received.

Reply:
Substrate gossip requires you to manually resend against the topic, if you want to propagate to the peers (we want to in case of time keeper). `ValidationResult::ProcessAndKeep` and `ValidationResult::Discard` only indicate if these should be kept in the per-peer queue or not(to validate future messages for duplicates, etc), from when I looked at their code recently. Pls correct if it is not so ..

Reply:
Yeah, I think you're write, this probably needs to be restored. At the same time why should this only be done by time keeper? Client should also re-gossip received messages, right?

Reply:
Consensus nodes already receive O(num time keeper) messages per second now which cannot be avoided. Would be good to avoid adding more volume to that. And if they gossip, will go back to time keepers also. We could possibly consider two different topics in future.

In case of time keepers, the gossip was needed for node sync IIRC. This part is still changing though

Reply:
I do not think that is correct. You assume that every consensus node is connected to at least one timekeeper. That is not guaranteed to be the case, very far from it actually.

Reply:
I see, I was assuming a close to full mesh topology. In that case we would indeed need to gossip from consensus nodes also